lib/node-extensions.js

```diff
@@ -22,7 +22,10 @@ Promise.denodeify = function (fn, argumentCount) {
         if (err) reject(err)
         else resolve(res)
       })
-      fn.apply(self, args)
+      var res = fn.apply(self, args)
+      if (res && res.then) {
```

Comment:
Should probably do the full check for a promise here:

```js
(res && (typeof res === 'object' || typeof res === 'function') && typeof res.then === 'function')
```

Guards against things that just look a little bit like a promise but aren't.